services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/functionaltests/AtomicCounterTest.java

```diff
@@ -112,15 +112,31 @@ public void createViaPut_incrementsCorrectly() {
     }
 
     @Test
-    public void createViaUpdate_settingCounterInPojo_throwsException() {
+    public void createViaUpdate_settingCounterInPojo_hasNoEffect() {
         AtomicCounterRecord record = new AtomicCounterRecord();
         record.setId(RECORD_ID);
         record.setDefaultCounter(10L);
         record.setAttribute1(STRING_VALUE);
 
-        assertThatThrownBy(() -> mappedTable.updateItem(record))
-            .isInstanceOf(DynamoDbException.class)
-            .hasMessageContaining("Two document paths");
+        mappedTable.updateItem(record);
+        AtomicCounterRecord persistedRecord = mappedTable.getItem(record);
+        assertThat(persistedRecord.getAttribute1()).isEqualTo(STRING_VALUE);
+        assertThat(persistedRecord.getDefaultCounter()).isEqualTo(0L);
+        assertThat(persistedRecord.getCustomCounter()).isEqualTo(10L);
+        assertThat(persistedRecord.getDecreasingCounter()).isEqualTo(-20L);
+    }
+
+    @Test
+    public void updateItem_retrievedFromDb_shouldNotThrowException() {
+        AtomicCounterRecord record = new AtomicCounterRecord();
+        record.setId(RECORD_ID);
+        record.setAttribute1(STRING_VALUE);
+        mappedTable.updateItem(record);
+
+        AtomicCounterRecord retrievedRecord = mappedTable.getItem(record);
+        retrievedRecord.setAttribute1("ChangingThisAttribute");
+
+        assertThat(mappedTable.updateItem(retrievedRecord)).isNotNull();
```

Comment:
Can we add an assertion for the counter value?

Reply:
yes